From adb92c7648ddf982e6e879cc5ca02522b89ab511 Mon Sep 17 00:00:00 2001
From: Huang Jiachai <hjc@rock-chips.com>
Date: Mon, 29 Feb 2016 10:33:25 +0800
Subject: [PATCH] video: rockchip: fb: fix update reg kfree data lead to
 list_del null point

Change-Id: I5a9f3c795cf6f8ecbe9f45a769135fe890525fab
Signed-off-by: Huang Jiachai <hjc@rock-chips.com>
(cherry picked from commit fe05c53704fb3b61455dddbae8b5078be0f48a12)
---
 drivers/video/rockchip/rk_fb.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/video/rockchip/rk_fb.c b/drivers/video/rockchip/rk_fb.c
index e8eeda5e0764..9fb1a5eeb20f 100644
--- a/drivers/video/rockchip/rk_fb.c
+++ b/drivers/video/rockchip/rk_fb.c
@@ -1940,8 +1940,8 @@ static void rk_fb_update_regs_handler(struct kthread_work *work)
 	mutex_unlock(&dev_drv->update_regs_list_lock);
 
 	list_for_each_entry_safe(data, next, &dev_drv->saved_list, list) {
-		rk_fb_update_reg(dev_drv, data);
 		list_del(&data->list);
+		rk_fb_update_reg(dev_drv, data);
 	}
 
 	if (dev_drv->wait_fs && list_empty(&dev_drv->update_regs_list))
-- 
2.35.3

